### ISSUE TITLE
Integration: Flakey actor tests

### ISSUE THREAD

**Issue by JoshVanL** [dapr/dapr#7250]
Currently, the integration tests are flaky and almost never pass on the MacOS runner. 

These errors are usually exclusively due to actor related tests, for example `Test_Integration/actors/reminders/rebalancing`:

```
2023-11-27T22:12:57.5972570Z 2023/11/27 22:12:43 Failed to report status to placement; will retry: error from placement: rpc error: code = FailedPrecondition desc = The cluster's Actor API level is 10, which is higher than the reported API level 0
```

Integration tests should be made stable/bugs fixed, ideally before any other actor related features are merged.

**Comment by ItalyPaleAle** [comment]
Fix is in #7129. Unclear why it only appears on macOS (probably due to some timings)

**Comment by mukundansundar** [comment]
@ItalyPaleAle can the fix for this be a separate PR? that will unblock the other PRs that target Actors

**Comment by ItalyPaleAle** [comment]
> @ItalyPaleAle can the fix for this be a separate PR? that will unblock the other PRs that target Actors 

The fix is in #7129 and the concerns @JoshVanL has to hold up the PR are about tests. So I can split those out but sounds like that's not enough.

**Comment by ItalyPaleAle** [comment]
> @ItalyPaleAle can the fix for this be a separate PR? that will unblock the other PRs that target Actors 

I'm happy to do that tomorrow. I can extract the test fixes from that PR. At this time the holdup was that @JoshVanL had concerns on the tests (the actual feature seemed fine based on the review) so I don't know if that will help much but i can do it anyways.
